internal/oci/store/store.go

```diff
@@ -346,3 +356,16 @@ func copyChunks(dst io.Writer, src io.Reader, chunkSize int64) (int64, error) {
 		}
 	}
 }
+
+// Validate returns an error if the supplied image is invalid,
+// e.g. the number of layers is above the maximum allowed.
```

Comment:
Maybe add a TODO mentioning that this _could_ be merged with go-containerregistry's `validate.Image` in future?